src/game/logic/object/object.cpp

```diff
@@ -732,3 +732,49 @@ bool Object::Set_Script_Status(ObjectScriptStatusBit bit, bool set)
     return false;
 #endif
 }
+
+bool Object::Is_Faction_Structure() const
+{
+    return Is_Any_KindOf(KINDOFMASK_FS);
+}
+
+ProductionUpdateInterface *Object::Get_Production_Update_Interface()
+{
+    for (BehaviorModule **module = m_allModules; *module != nullptr; module++) {
+        ProductionUpdateInterface *update = (*module)->Get_Production_Update_Interface();
+
+        if (update != nullptr) {
+            return update;
+        }
+    }
+
+    return nullptr;
+}
+
+void Object::Set_Vision_Spied(bool vision_spied, int player)
+{
+    bool changed = false;
```

Comment:
No assert here to verify player index bounds?

Reply:
nope.